frontend/src/libraries/explorer-wamp/transactions.ts

```diff
@@ -222,109 +222,112 @@ export default class TransactionsApi extends ExplorerApi {
-        if (
-          receipts.length === 0 ||
-          receipts[0].receipt_id !== receiptsOutcome[0].id
-        ) {
-          receipts.unshift({
-            predecessor_id: transactionExtraInfo.transaction.signer_id,
-            receipt: actions,
-            receipt_id: receiptsOutcome[0].id,
-            receiver_id: transactionExtraInfo.transaction.receiver_id,
-          });
-        }
-        const receiptOutcomesByIdMap = new Map();
-        receiptsOutcome.forEach((receipt: any) => {
-          receiptOutcomesByIdMap.set(receipt.id, receipt);
-        });
-
-        const receiptsByIdMap = new Map();
-        receipts.forEach((receiptItem: any) => {
-          if (receiptItem.receipt_id === receiptsOutcome[0].id) {
-            receiptItem.actions = actions;
-          } else {
-            const { Action: action = undefined } = receiptItem.receipt;
-            receiptItem.actions = action?.actions.map(
-              (action: RpcAction | string) => {
-                if (typeof action === "string") {
-                  return { kind: action, args: {} };
-                } else {
-                  const kind = Object.keys(action)[0] as keyof RpcAction;
-                  return {
-                    kind,
-                    args: action[kind],
-                  };
-                }
-              }
-            );
-          }
-          receiptsByIdMap.set(receiptItem.receipt_id, receiptItem);
-        });
-
-        const collectNestedReceiptWithOutcome = (receiptHash: string) => {
-          const receipt = receiptsByIdMap.get(receiptHash);
-          const receiptOutcome = receiptOutcomesByIdMap.get(receiptHash);
-          return {
-            ...receipt,
-            ...receiptOutcome,
-            outcome: {
-              ...receiptOutcome.outcome,
-              outgoing_receipts: receiptOutcome.outcome.receipt_ids.map(
-                (executedReceipt: string) =>
-                  collectNestedReceiptWithOutcome(executedReceipt)
-              ),
-            },
-          };
-        };
-
-        transactionInfo.actions = actions;
-        transactionInfo.receiptsOutcome = receiptsOutcome;
-        transactionInfo.receipt = collectNestedReceiptWithOutcome(
-          receiptsOutcome[0].id
-        ) as NestedReceiptWithOutcome;
-        transactionInfo.transactionOutcome = transactionExtraInfo.transaction_outcome as TransactionOutcome;
-      }
-      return transactionInfo;
     } catch (error) {
       console.error(
         "TransactionsApi.getTransactionInfo failed to fetch data due to:"
       );
       console.error(error);
       throw error;
     }
+
+    if (!transactionInfo) {
+      throw new Error(
+        `TransactionsApi.getTransactionInfo: transaction '${transactionHash}' not found`
+      );
+    } else {
+      const transactionExtraInfo = await this.call<any>("nearcore-tx", [
+        transactionInfo.hash,
+        transactionInfo.signerId,
+      ]);
```

Comment:
This might raise exceptions either due to WAMP being unavailable or RPC node behind `nearcore-tx` being unavailable, etc. We should handle those cases gracefully.